src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -2273,6 +2275,21 @@ private static void CheckParameterModifierMismatchMethodConversion(SyntaxNode sy
                 return;
             }
 
+            // For synthesized delegates with mismatched escape rules, report an error to prevent
+            // ref-safety holes across old and new rule boundaries.
+            // See https://github.com/dotnet/roslyn/issues/76087
+            if (isSynthesizedDelegate &&
+                delegateMethod is not null &&
+                delegateMethod.UseUpdatedEscapeRules != lambdaOrMethod.UseUpdatedEscapeRules &&
+                SourceMemberContainerTypeSymbol.RequiresValidScopedOverrideForRefSafety(delegateMethod, lambdaOrMethod.TryGetThisParameter(out var methodThisParam) ? methodThisParam : null))
```

Comment:
I'm not following how is `RequiresValidScopedOverrideForRefSafety` related to the issue.